### PR TITLE
refactor(app): update modules interfaces

### PR DESCRIPTION
**Pull request by sakibh**
# Overview

This PR updates the interfaces of the modules to reflect the shape of the data returned by the API.

# Changelog

- Updated interfaces for the modules
- Deleted some legacy code

# Review requests

- Check that the interfaces match the shape of the data being returned by the API.
- Check that module controls are now able to post commands.

# Risk assessment

updates module interfaces and fixtures

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/SetupModules.tsx [hunk 1]
```diff
@@ -77,7 +77,7 @@ export const SetupModules = ({
   )
 
   const hasADuplicateModule = new Set(moduleModels).size !== moduleModels.length
-
+  console.log(heaterShakerModules[0]?.moduleDef.displayName)
```
Comment (smb2268) [inline app/src/organisms/Devices/ProtocolRun/SetupModules.tsx]: left behind console.log statement

Comment (koji) [inline app/src/organisms/Devices/ProtocolRun/SetupModules.tsx]: Do you need to keep this `console.log`?

**Review by smb2268 — APPROVED** [review]
This looks good to me! Is there any concern of backwards compatibility issues or is that taken care of with the HTTP_API_VERSION?

**Review by koji — APPROVED** [review]
left two comments
Looks good to me 🚀

**Review by jerader — APPROVED** [review]
lgtm! Also did some smoke testing, really cool to see the commands work ⭐ thank you for updating these!

File: app/src/redux/modules/api-types.ts [hunk 2]
```diff
@@ -9,66 +14,68 @@ import type {
...
   moduleModel: ModuleModel
-  fwVersion: string
-  port: string
+  moduleType: ModuleType
+  firmwareVersion: string
   hasAvailableUpdate: boolean
   usbPort: PhysicalPort
 }
 
 interface ApiBaseModuleLegacy {
```
Comment (b-cooper) [inline app/src/redux/modules/api-types.ts:33]: I believe the legacy api module type and logic can be deleted now.  Don't worry about it for this PR, but it is worth making a follow-up ticket to throw on the backlog and clean out these unused types.

**Review by b-cooper — APPROVED** [review]
🍧 looks great!

File: app/src/redux/modules/api-types.ts [hunk 3]
```diff
@@ -9,66 +14,68 @@ import type {
...
 }
 export interface HeaterShakerData {
   labwareLatchStatus: LatchStatus
   speedStatus: SpeedStatus
   temperatureStatus: TemperatureStatus
   currentSpeed: number | null
-  currentTemp: number | null
+  currentTemperature: number | null
   targetSpeed: number | null
   targetTemp: number | null
```
Comment (koji) [inline app/src/redux/modules/api-types.ts]: Is there any reason you avoid changing this to `targetTemperature`?

**Pull request merged**